controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java

```diff
@@ -127,6 +130,11 @@ public boolean authorize(String resource, Principal principal, String credential
         return handler.authorize(resource, principal).ordinal() >= level.ordinal();
     }
 
+    @VisibleForTesting
+    public synchronized void registerHandler(AuthHandler authHandler) {
+        this.handlerMap.put(authHandler.getHandlerName(), authHandler);
```

Comment:
It's meant to be used for unit testing only. In RESTAuthHelperTest I'm registering a FakeAuthHandler so that I unit test auth functionality in RESTAuthHelper.

Reply:
If this map is only for testing, then it should be located in test code. The best way to do this is to wrap or subclass the real class in the test class and inject the modified implementation.

Reply:
The map `handlerMap` itself is used in the main code as well. It holds entries comprising of custom auth handlers by handler name. The method `registerHandler(..)` was the one that was added to aid unit testing, in order to make it possible to inject an auth handler entry into `PravegaAuthManager`.  
I took a cue from your suggestion to also explore if it'd be useful to create an inherited sub-class which adds this sole method `registerHandler`, but then felt it'd be overkill in this case. Besides, having this method right here makes it easier for others to spot it and use it in their own tests when necessary. A comment might be useful here - I'll add it.
Please let me know if you agree.